reference/security.rst

```diff
@@ -0,0 +1,252 @@
+JAAS Security Overview
+=======================
+
+This document provides an overview of JAAS security measures, focusing on areas related to 
+sensitive data storage, transmission, and cryptographic technologies.
+
+Cloud Credentials
+-----------------
+
+Cloud credentials are API keys used by Juju/JAAS to provision cloud infrastructure. 
+These credentials are securely uploaded to and stored by JAAS. 
+When a model is created, the key is passed to the Juju controller to manage cloud resources.
+
+User-provided cloud credentials are stored securely in a `Vault <https://www.vaultproject.io/>`__, 
+a tool for managing secrets. Ensuring the secure handling of these credentials is essential 
+to prevent unauthorised access or data breaches.
+
+Vault encrypts credentials at rest and provides mechanisms to prevent unauthorised access.
+
+JAAS - Juju Communication
+--------------------------
+
+JAAS acts as an authentication gateway between users and Juju controllers. 
+Juju controllers trust JAAS by setting the ``login-token-refresh-url`` during bootstrap.
+More information on setting up a Juju controller for JAAS can be found in our :doc:`how-to guide <../how-to/add_controller>`.
+
+Trust between Juju controllers and JAAS is established through asymmetric cryptography 
+and `JSON Web Tokens (JWTs) <https://jwt.io/introduction>`__.
+
+.. hint::
+    While JWTs are also used for user sessions, the tokens exchanged between JAAS and Juju controllers are separate from user tokens.
+
+The ``login-token-refresh-url`` points to a 
+`JSON Web Key Set (JWKS) <https://auth0.com/docs/secure/tokens/json-web-tokens/json-web-key-sets>`__
+endpoint, containing public keys used to verify tokens issued by JAAS.
+
+The public and private key pair are generated at startup and placed in Vault.
+The keys are then periodically rotated.
+
+When JAAS makes requests to a Juju controller, a JWT is issued that encodes user information 
+and permissions, securely delegating authentication and authorisation from the controller to JAAS.
+
+Details:
+
+- **JWKS Endpoint**: ``<jimm-url>/.well-known/jwks.json``
+- **Key Type/Size**: RSA 4096 bits
+- **Signing Algorithm**: RS256
+
+The following Go packages are used:
+
+- ``github.com/lestrrat-go/jwx/v2/jwa``
+- ``github.com/lestrrat-go/jwx/v2/jwt``
+- ``github.com/lestrrat-go/jwx/v2/jwk``
+
+User Sessions
+-------------
+
+CLI-Based Sessions
+^^^^^^^^^^^^^^^^^^
+
+When authenticating the Juju CLI to JAAS, users go through an OAuth login flow, 
+after which the CLI is issued a JWT session token. This token is stored on the 
+filesystem and sent with each request, eliminating the need for repeated logins.
+
+While Juju controller tokens use asymmetric cryptography, session tokens for 
+users are signed using symmetric cryptography (i.e., a shared secret) with a 
```

Comment:
Let's explain why its symmetric,

We use symmetric cryptography for application sessions as to prevent the user decoding the token (I know we only sign it, but we should encrypt it). Only JIMM needs to know the contents of the token.